source/validate_id.cpp

```diff
@@ -300,6 +302,31 @@ bool idUsage::isValid<SpvOpEntryPoint>(const spv_instruction_t* inst,
       return false;
     }
   }
+
+  std::stack<uint32_t> call_stack;
+  call_stack.push(entryPoint->id());
+  while (!call_stack.empty()) {
```

Comment:
Recommend using a visited set to avoid cycling if there are cycles in the call graph.

That's possible to express in SPIR-V, but banned by all the known client APIs.  But the validator should not loop forever on such a bad input.